mods/ra/maps/soviet-05/soviet05-AI.lua

```diff
@@ -20,64 +21,6 @@ IdlingUnits = function()
 	end)
 end
 
-BaseBuildings =
-{
-	{ type = "powr", pos = CVec.New(3, -2), cost = 300 },
```

Comment:
Were these offsets taken from the original mission (which IIRC did use predefined buildings and positions), or did we make them up?

I haven't tested the mission yet, but i'm going to be surprised if HackyAI's building fraction+placement logic isn't going to be worse in practice than our predefined list. I'm expecting that a good compromise will be to have some kind of `ScriptedBaseBuilderBotModule` that takes a list of actor types and positions similar to this.

Reply:
I don't remember. Personally I don't care about giving buildings exact locations. The AI didn't build completely stupid in my tests and I'd rather improve our AI logic (better distribution for defences for example) than introduce more maintenance burden by adding logic for predefined positions.

The general idea was to remove much of the hardcoded AI scripting (except say for some special waves and the likes, always depending on the case) and replace it by our default AI logic, profiting from the planned improvements and from those already in the queue, to make it feel more natural than the hardcoded AI.

Reply:
My concern is that many missions have very constrained base layouts with walls and lanes for units to move through. We don't want the AI to start haphazardly blocking the gates with base defenses or base-walking power plants outside the walls!